File: Java/Algorithms and Data Structures/String Algorithms/ReverseString.java

public class ReverseString {
    public static String reverseString(String args) {
        char strLetter = (char)0;
        StringBuilder sb = new StringBuilder();
        String reverseOut = "";
        for(int i= args.length() - 1;i >= 0;i--){
            strLetter = args.charAt(i);
            reverseOut = sb.append(strLetter).toString();
        }
        System.out.println(reverseOut);
        return reverseOut;
    }
    public static void main(String[] args) {
        String input = "lilac";
        reverseString(input);
    }
}

File: Python/Algorithms and Data Structures/String Algorithms/ReverseString.py

def reverseString(input_string):
    reverse_string = ""
    char_at_i = ''
    for i in (len(input_string) - 1,0,-1):
        char_at_i = input_string[i]
        reverse_string = reverse_string + char_at_i
    print(reverse_string)

#input_string = input("Enter the input String: ")
input_string = 'jef'
reverseString(input_string)
